raigad/src/main/java/com/netflix/raigad/monitoring/HealthMonitor.java

```diff
@@ -111,6 +110,7 @@ else if (clusterHealthStatus.name().equalsIgnoreCase("RED")) {
             logger.warn("failed to load Cluster Health Status", e);
         }
 
+        healthBean.esuponinstance = ElasticsearchProcessMonitor.isElasticsearchStarted() ? 1 : 0;
```

Comment:
This mean that we should modify the initial check:

        // If Elasticsearch is started then only start the monitoring
        if (!ElasticsearchProcessMonitor.isElasticsearchStarted()) {
            String exceptionMsg = "Elasticsearch is not yet started, check
back again later";
            logger.info(exceptionMsg);
            return;
        }


bailing out should be done only while ES is restarting (node is
bootstrapping), but after it's done - we should never fail.
Regarding alerts, we use rolling-count in order to prevent transient errors
from alerting, meaning, if you restart an instance and it was down for 3
minutes - an alerts shouldn't get triggered if we used a rolling count of
7/10 minutes.

Currently, we're not sending any metric when ES process is down - and
that's a problem if we want to create alerts based on these metrics - no?

Is there a way to "ask" ES if it's initializing/starting (like we can see
with shards) ?





On Thu, Apr 16, 2015 at 2:16 PM, Sagar Loke <notifications@github.com>
wrote:

> In raigad/src/main/java/com/netflix/raigad/monitoring/HealthMonitor.java
> <https://github.com/Netflix/Raigad/pull/26#discussion_r28550688>:
>
> > @@ -111,6 +110,7 @@ else if (clusterHealthStatus.name().equalsIgnoreCase("RED")) {
> >              logger.warn("failed to load Cluster Health Status", e);
> >          }
> >
> > +        healthBean.esuponinstance = ElasticsearchProcessMonitor.isElasticsearchStarted() ? 1 : 0;
>
> This check will always be true. Meaning, here you won't be able to catch
> if ES is not running, because of Initial Check in the beginning of
> execute() method.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Netflix/Raigad/pull/26/files#r28550688>.
>